### PR TITLE
Migrate Beam benchmark to use Gradle for Java runs

### PR DESCRIPTION
**Pull request by lgajowy**
Since there is an ongoing maven -> gradle migration in beam, the mvn commands located in beam_benchmark_helper.py should be migrated to gradle's equivalents.

More on that matter here: https://issues.apache.org/jira/browse/BEAM-3942. The beam's part was done, now it's time to change what's necessary in PerfKit.

@asaksena could you take a look? If you don't mind, I'd also like @DariuszAniszewski to take a look before merging this, as he also works very closely with this benchmark.

### CODE REVIEW

File: perfkitbenchmarker/beam_benchmark_helper.py [hunk 1]
```diff
@@ -275,10 +290,13 @@ def _BuildPythonCommand(benchmark_spec, modulename, job_arguments):
   return cmd
 
 
+def _GetGradleCommand():
+  return FLAGS.gradle_binary if FLAGS.gradle_binary else _GetBeamDir() + '/gradlew'
+
 def _GetBeamDir():
   # TODO: This is temporary, find a better way.
   return FLAGS.beam_location if FLAGS.beam_location else os.path.join(
```
Comment (yuyantingzero) [inline perfkitbenchmarker/beam_benchmark_helper.py]: return FLAGS.beam_location or os.path.join(...)

File: perfkitbenchmarker/beam_benchmark_helper.py [hunk 2]
```diff
@@ -275,10 +290,13 @@ def _BuildPythonCommand(benchmark_spec, modulename, job_arguments):
   return cmd
 
 
+def _GetGradleCommand():
+  return FLAGS.gradle_binary if FLAGS.gradle_binary else _GetBeamDir() + '/gradlew'
```
Comment (yuyantingzero) [inline perfkitbenchmarker/beam_benchmark_helper.py]: return FLAGS.gradle_binary or _GetBeamDir() + '/gradlew'

File: perfkitbenchmarker/beam_benchmark_helper.py [hunk 3]
```diff
@@ -192,44 +217,34 @@ def _BuildMavenCommand(benchmark_spec, classname, job_arguments):
...
-                              FLAGS.beam_runner_profile)
-  AddRunnerOptionMvnArgument(benchmark_spec.service_type, beam_args,
-                             FLAGS.beam_runner_option)
-  AddExtraMvnProperties(cmd, FLAGS.beam_extra_mvn_properties)
+  AddRunnerArgument(cmd)
+  AddRunnerPipelineOption(beam_args)
+  AddFilesystemArgument(cmd)
+  AddExtraProperties(cmd)
 
   cmd.append("-DintegrationTestPipelineOptions="
```
Comment (yuyantingzero) [inline perfkitbenchmarker/beam_benchmark_helper.py]: be consistent about quotes. Stick with ' ?

File: perfkitbenchmarker/beam_benchmark_helper.py [hunk 4]
```diff
@@ -192,44 +217,34 @@ def _BuildMavenCommand(benchmark_spec, classname, job_arguments):
...
              "[{}]".format(','.join(beam_args)))
 
+  cmd.append("--stacktrace")
```
Comment (yuyantingzero) [inline perfkitbenchmarker/beam_benchmark_helper.py]: be consistent with quotes

File: perfkitbenchmarker/beam_benchmark_helper.py [hunk 5]
```diff
@@ -180,8 +205,8 @@ def BuildBeamCommand(benchmark_spec, classname, job_arguments):
   return cmd, base_dir
 
 
-def _BuildMavenCommand(benchmark_spec, classname, job_arguments):
-  """ Constructs a maven command for the benchmark.
+def _BuildGradleCommand(classname, job_arguments):
+  """ Constructs a Gradle command for the benchmark.
 
   Args:
     benchmark_spec: The PKB spec for the benchmark to run.
```
Comment (yuyantingzero) [inline perfkitbenchmarker/beam_benchmark_helper.py]: remove benchmark_spec?

File: perfkitbenchmarker/beam_benchmark_helper.py (hunk 2 above)
Comment (DariuszAniszewski) [reply perfkitbenchmarker/beam_benchmark_helper.py]: you may also use `os.path.join` to build `gradlew` path

File: perfkitbenchmarker/beam_benchmark_helper.py (hunk 1 above)
Comment (lgajowy) [reply perfkitbenchmarker/beam_benchmark_helper.py]: ok

File: perfkitbenchmarker/beam_benchmark_helper.py (hunk 2 above)
Comment (lgajowy) [reply perfkitbenchmarker/beam_benchmark_helper.py]: ok

File: perfkitbenchmarker/beam_benchmark_helper.py (hunk 3 above)
Comment (lgajowy) [reply perfkitbenchmarker/beam_benchmark_helper.py]: ok

File: perfkitbenchmarker/beam_benchmark_helper.py (hunk 4 above)
Comment (lgajowy) [reply perfkitbenchmarker/beam_benchmark_helper.py]: ok

File: perfkitbenchmarker/beam_benchmark_helper.py (hunk 5 above)
Comment (lgajowy) [reply perfkitbenchmarker/beam_benchmark_helper.py]: ok

**Comment by lgajowy** [comment]
Thanks, I applied your suggestions @yuyantingzero. 
Could you guys take another look?

**Comment by lgajowy** [comment]
@asaksena @yuyantingzero do you think this contribution is ready to merge then?

**Comment by lgajowy** [comment]
@asaksena @yuyantingzero merging this is crucial to finish up the maven -> gradle migration in beam. Could you please take a look?

Documentation proving that beam uses gradle now is here: https://beam.apache.org/documentation/io/testing/

**Comment by asaksena** [comment]
I have approved the changes, but some tests are failing. Can you please fix those ?
For eg.

ERROR: test_extra_mvn_property_empty_property (tests.beam_benchmark_helper_test.BeamBenchmarkHelperTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/GoogleCloudPlatform/PerfKitBenchmarker/tests/beam_benchmark_helper_test.py", line 90, in test_extra_mvn_property_empty_property
    beam_benchmark_helper.AddExtraMvnProperties(actual_mvn_command, testOptionVal)
AttributeError: 'module' object has no attribute 'AddExtraMvnProperties'

**Comment by lgajowy** [comment]
@asaksena thank you and sorry for missing the build break (my fault!). I fixed the tests and adapted them to the current benchmark functionality. Could you take another look?

**Comment by asaksena** [comment]
Merged

**Pull request merged**